docs/src/generating-a-schema/index.md

```diff
@@ -0,0 +1,155 @@
+# Automatically generating GraphQL schema from JSON ABI
+
+`forc index new` supports automatically generating GraphQL schema from a contract JSON ABI.
```

Comment:
- I think this has to be re-written. 
- @lostman We spent _weeks_ on the docs, combing through every single line making sure they read properly and that all examples are informative and work
  - It sucked, it wasn't fun, but the docs are _so_ much better now
  - Let's keep this trend going with the docs in this PR.
  - Ideally the docs should:
    1. Show me the tiniest Sway smart contract with various simple types (struct, enum, etc)
    2. Show me how to use `forc index new --json-abi`
    3. Show me the resultant schema that gets produced from that command ☝🏼 
  - I think we have an example of this in the docs already [here](https://docs.fuel.network/docs/indexer/indexing-custom-types/) (again, I want to be very clear that I realize this is a PITA 😅 )